Mathlib/Data/Fin/Tuple/Basic.lean

```diff
@@ -410,6 +411,41 @@ theorem append_castAdd_natAdd {f : Fin (m + n) → α} :
   unfold append addCases
   simp
 
+theorem append_injective_iff {α : Type*} {m n : ℕ} {x : Fin m → α} {y : Fin n → α} :
```

Comment:
```suggestion
theorem append_injective_iff {x : Fin m → α} {y : Fin n → α} :
```
I think the variable is already there

Reply:
No, its a `Sort` valued and `Set.range` doesn't want of that.

Reply:
Perhaps worth replacing `Disjoint (Set.range x) (Set.range y)` with `∀ i j, x i ≠ y j` then?

Reply:
Apart from this comment LGTM. @eric-wieser is it OK fro you?

Reply:
I'd like to see if I can make the golf in https://github.com/leanprover-community/mathlib4/pull/24167#discussion_r2088668707 work once the linked PR lands